src/transformers/models/bark/modeling_bark.py

```diff
@@ -0,0 +1,1372 @@
+
+        hidden_states = self.transformer.drop(input_embeds + position_embeds)
+        output_shape = input_shape + (hidden_states.size(-1),)
+
+        all_self_attentions = () if output_attentions else None
+        all_hidden_states = () if output_hidden_states else None
+
+        for i, block in enumerate(self.transformer.h):
+            if output_hidden_states:
+                all_hidden_states = all_hidden_states + (hidden_states,)
+
+            outputs = block(
+                hidden_states,
+                attention_mask=attention_mask,
+                head_mask=head_mask[i],
+                output_attentions=output_attentions,
+            )
+
+            hidden_states = outputs[0]
+
+            if output_attentions:
+                all_self_attentions = all_self_attentions + (outputs[1],)
+
+        hidden_states = self.transformer.ln_f(hidden_states)
+        hidden_states = hidden_states.view(output_shape)
+
+        # Add last hidden state
+        if output_hidden_states:
+            all_hidden_states = all_hidden_states + (hidden_states,)
+
+        logits = self.lm_heads[pred_idx - self.config.n_codes_given](hidden_states)
+
+        loss = None
+        if labels is not None:
+            raise NotImplementedError("Training is not implemented yet")
+
+        if not return_dict:
+            return tuple(v for v in [None, logits, all_hidden_states, all_self_attentions] if v is not None)
+
+        return MaskedLMOutput(
+            loss=loss,
+            logits=logits,
+            hidden_states=all_hidden_states,
+            attentions=all_self_attentions,
+        )
+
+
+@add_start_docstrings(
+    """
+    HugginFace implementation of `Bark`, a text-to-speech model composed of 3 sub-models:
+    1. A semantic model, that refines the input information.
+    2. A coarse acoustics model, that generates a raw version of the audio with the two first codebooks of `Encodec`.
+    3. A fine acoustics model, that refines the raw audio by generating the last 6 codebooks of `Encodec`, following
+       `Bark` original implementation.
+    """,
+    BARK_START_DOCSTRING,
+)
+class BarkModel(BarkPreTrainedModel):
+    # TODO: Add code sample when preprocessing is clearer, and encodec is added.
+    # @add_code_sample_docstrings(
+    #    checkpoint=_CHECKPOINT_FOR_DOC,
+    #    output_type=BaseModelOutputWithPast,
+    #    config_class=_CONFIG_FOR_DOC,
+    # )
+    _no_split_modules = ["BarkBlock"]
+
+    def __init__(self, config):
+        super().__init__(config)
+
+        self.semantic_model = BarkCausalModule(config.semantic_config)
+        self.coarse_acoustics_model = BarkCausalModule(config.coarse_acoustics_config)
+        self.fine_acoustics_model = BarkFineAcousticsModule(config.fine_acoustics_config)
+
+        # TODO: for now, it is not integrated into the code
+        self.codec_model = EncodecModel.encodec_model_24khz()
+        self.codec_model.set_target_bandwidth(6.0)
```

Comment:
I think our API will call the bandwidth in the forward call:
```python
model.decode(codes, bandwith=6.0)
```
Or use the lowest bandwidth from the config (`model.config.bandwidth[0]`)

Reply:
I've modified it in the new commits. I think that `bandwidth` is not used in the `encodec.quantizer.decode` and `encodec.decoder` but I might be wrong, can you verify for double check?